one-ui/src/main/java/com/marklogic/hub/oneui/auth/AuthenticationFilter.java

```diff
@@ -137,15 +139,24 @@ protected AuthenticationToken authenticateUser(String username, String password)
      * - see https://docs.spring.io/spring-security/site/docs/current/reference/htmlsingle/#appendix-faq-role-prefix.
      *
      * @param stagingClient
+     * @param roles
+     * @param canInstallDataHub
      * @return
      */
-    protected List<GrantedAuthority> getAuthoritiesForAuthenticatedUser(DatabaseClient stagingClient) {
+    protected List<GrantedAuthority> getAuthoritiesForAuthenticatedUser(DatabaseClient stagingClient, ArrayNode[] roles, boolean[] canInstallDataHub) {
```

Comment:
When we need a method to return a tuple of different things, I'd much rather see us use the Pair class or a custom class instead of passing arrays as ways to capture values. 

So this can be simplified by returning a Pair<List\<GrantedAuthority\>, ArrayNode> . That makes it much more obvious what the method is doing.

For "canInstallDataHub" - can't LoginHandler just iterate over the list of GrantedAuthoritys to see if "canInstallDataHub" exists, and if so, write that to the JSON response? It shouldn't need to care about hasManagePrivileges anymore, right? That would get rid of this from this function, which shouldn't care about setting bits specific to the login response. It should just care about getting all the authorities (and for now, all the roles too, though that will soon go away).

Reply:
Now UI needs hasManagePrivileges for convenience. We can remove it once UI do not need it. Since we will remove roles, we can refactor the codes later on.

Reply:
I do not see any difference between use a Pair as return and pass roles as a parameter...

Reply:
It's more clear what the inputs and outputs are. A signature of:

    List<GrantedAuthority> getGrantedAuthorities(DatabaseClient, ArrayNode); 

implies that the DatabaseClient and ArrayNode are inputs needed to process a single List as the output. The reader has to do extra work to realize that one of the inputs is actually an output, not an input. 

But this signature:

    Pair<List, ArrayNode> getGrantedAuthorities(DatabaseClient);

implies that a single input is needed to produce two outputs. 

But it's fine now not to change it - ideally, the roles will go away soon and we'll have a single output - the List of authorities.